5.x/crud-operation-trash.md

```diff
@@ -0,0 +1,192 @@
+# Trash Operation <span class="badge badge-info">PRO</span>
+
+--
+
+<a name="about"></a>
+## About
+
+This CRUD operation allows your admins to soft delete, restore and permanently delete entries from the table.
+
+<a name="requirements"></a>
+## Requirements
+
+1. This is a <span class="badge badge-info">PRO</span> operation. It requires that you have [purchased access to `backpack/pro`](https://backpackforlaravel.com/products/pro-for-unlimited-projects).
+
+2. In addition, it needs that your Model uses Laravel's `SoftDeletes` trait. To have your Model use SoftDeles, you should:
+- generate a migration to add the `deleted_at` column to your table, eg. `php artisan make:migration add_soft_deletes_to_products --table=products`;
+- inside that file's `Schema::table()` closure, add `$table->softDeletes();`
+- run `php artisan migrate`
+- add `use SoftDeletes` on the corresponding model (and import that class namespace);  
+
+<a name="trash-a-single-item"></a>
+## Trash a Single Item <span class="badge badge-info">PRO</span>
+
+<a name="how-it-works"></a>
+### How it Works
+- **Trash**:
+Using AJAX, a DELETE request is performed towards ```/entity-name/{id}/trash```, which points to the ```trash()``` method in your EntityCrudController.
+
+- **Restore**:
+Using AJAX, a PUT request is performed towards ```/entity-name/{id}/restore```, which points to the ```restore()``` method in your EntityCrudController.
+
+- **Delete**:
+Using AJAX, a DELETE request is performed towards ```/entity-name/{id}/delete-permanently```, which points to the ```deletePermanently()``` method in your EntityCrudController.
+
+<a name="enabling"></a>
+### How to Use
+
+To enable it, you need to ```use \Backpack\Pro\Http\Controllers\Operations\TrashOperation;``` on your EntityCrudController. For example:
+
+```php
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use Backpack\CRUD\app\Http\Controllers\CrudController;
+
+class ProductCrudController extends CrudController
+{
+    use \Backpack\Pro\Http\Controllers\Operations\TrashOperation;
+}
+```
+
+This will make a Trash button and Trashed filter show up in the list view, and will enable the routes and functionality needed for the operation.
```

Comment:
```suggestion
**Step 1.** If your EntityCrudController uses the `DeleteOperation`, remove it. `TrashOperation` is a complete alternative to Delete.

**Step 2.** You need to ```use \Backpack\Pro\Http\Controllers\Operations\TrashOperation;``` inside your EntityCrudController. For example:

```php
class ProductCrudController extends CrudController
{
    // ... other operations
    use \Backpack\Pro\Http\Controllers\Operations\TrashOperation;
}```
This will make a Trash button and Trashed filter show up in the list view, and will enable the routes and functionality needed for the operation. If you're getting a "Trait not found" exception, make sure in the namespace you have typed `Backpack\Pro`, not `Backpack\PRO`. 
```